docs/Parser.puml

```diff
@@ -0,0 +1,26 @@
+@startuml
+'https://plantuml.com/sequence-diagram
+
+actor User
+
+User -> Main: fullUserInput
+activate Main
+Main -> Parser: parseCommand(fullUserInput)
+activate Parser
+
+alt COMMAND_WORD
+Parser -> Parser: prepareCommand(String requiredArguments)
```

Comment:
Perhaps have the arguments written as:

```suggestion
Parser -> Parser: prepareCommand(requiredArguments: String)
```

Reply:
wait sorry change of thought, I'm not sure what this should be written as because different prepareCommands take in different types of arguements